web: search select with focus, autocomplete, and progressive search

kensternberg-authentik opened the pull request:
web: search select with focus and progressive search

### What:

- New component: ak-list-select, which allows you to select from a list of elements, with keyboard control.
- New component: ak-portal, which manages elements by moving “slotted” content into a distant component, usually one attached to the body, and positions it relative to an existing element.
- ak-search-select-view has been revamped to handle focus, change, input, and blur using the browser native event handlers, rather than inventing my own.
- ak-search-select has been turned into a simple driver that manages the view.
- ak-search-select has a new declarative syntax for the most common use case.

### Why:

I seriously over-engineered this thing, leaning too heavily on outdated knowledge or assumptions about how the browser works. The native event handlers attached at the component’s borders works more than fine, and by attaching the event handlers to the portaled component before sending it off to the slots, the correct handlers get the message. This revision leverages the browser a *lot* more, and gets much more effective interaction with much less code.

### Details

#### ak-list-select

`<ak-list-select>` is a new component that replaces the ad-hoc menu object of the old SearchSelect. It is a standalone component that just shows a list, allows someone to navigate that list with the keyboard or the mouse. By default, it is limited to half the height of the viewport.

The list does not have an indicator for “selected” at this time. That’s just a side effect of it being developed as an adjunct to search-select. Its design does not preclude extension.

It has a *lot* of CSS components that can be customized. The properties and events are documented, but there is only one event: `change`. Consistent with HTML, the value is not sent with the `change` event; clients are expected to extract it with `change:event.target.value`.

Like all HTML components, it is completely stringly defined; the value is either a string or undefined.

#### ak-portal

`<ak-portal>` is a somewhat specialized “portal” component that places an `ak-list-select` in an object on top of the existing DOM content. It can be generalized to do this with any component, though, and can be extended. It has no events or CSS, since it’s “just” managing the portaling relationship.

#### ak-search-select-view

`<ak-search-select-view>` is the heart of the system. It takes a collection options and behaves like an autocomplete component for them. The only unique event it sends out is `change`, and like `ak-list-select`, it expects the client to retrieve the value.

Like all HTML components, it is completely stringly defined; the value is either a string or undefined.

#### ak-search-select

This is the SearchSelect component we’ve all known to come and love, but with a better pop-up and cleaner keyboard interaction. It emits only one event, `ak-change`, which *does* carry the value with it.

### Storybook & Tests

The Storybooks have been updated to show the current version of Search Select, with a (simulated) API layer as well as more blunt stringly-typed tests for the View layer. A handful of tests have been provided to cover a number of edge cases that I discovered during testing. These run fine with the `npx` command, and I would love to see them integrated into CI/CD.

### Developer Experience Enhancement!

The search select fields `renderElement`, `renderDescription`, and `value` properties of `ak-search-select` have been modified to take a string. For example, the search for the list of user looks like this:

    <ak-search-select
        .fetchObjects=${async (query?: string): Promise<User[]> => {
            const args: CoreUsersListRequest = { ordering: "username" };
            if (query !== undefined) {
                args.search = query;
            }
            const users = await new CoreApi(DEFAULT_CONFIG).coreUsersList(args);
            return users.results;
        }}
        .renderElement=${(user: User): string => {
            return user.username;
        }}
        .renderDescription=${(user: User): TemplateResult => {
            return html`${user.name}`;
        }}
        .value=${(user: User | undefined): string | undefined => {
            return user?.username;
         }}
    ></ak-search-select>

Since the Web Component API for integrating with the authentik TS API is a uniform collection of fields that can't be made into attributes, I've merged them into a single configuration object, which I think is easier to read:

    <ak-search-select-ez
        .config=${{
            fetchObjects: async (query?: string): Promise<Group[]> => {
                const args: CoreGroupsListRequest = {
                    ordering: "name",
                    includeUsers: false,
                    ... query ? { search: query } : {}
                };
                return (await new CoreApi(DEFAULT_CONFIG).coreGroupsList(args)).results;
             },
            renderElement: (group: Group): string => group.name,
            value: (group: Group | undefined): string | undefined => group?.pk,
            selected: (group: Group): boolean => group.pk === this.instance?.group
         }}
        blankable
    >
    </ak-search-select-ez>
    
It's possible to tighten up the `fetch` event further:

            fetchObjects: async (search?: string): Promise<Group[]> =>
                (
                    await new CoreApi(DEFAULT_CONFIG).coreGroupsList({
                        ordering: "name",
                        includeUsers: false,
                        ...(search ? { search } : {}),
                    })
                ).results,


-   [X] The code has been formatted (`make web`)

If applicable

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

kensternberg-authentik committed:
web: much better focus discipline

Fix the way focus is handled in SearchSelect so that the drop-down isn't grabbing the focus away
from the Input when the user wants to type in their selection.

Because it was broken otherwise!

There's still a bug where it's possible to type in a complete value
*Label*, then leave the component's focus (input and menu) completely,
in which case the Label remains, looking innocent and correct, but
it is *not* reflective of the value as understood by the SearchSelect
API controller.

Gonna try to fix that next.  But I'm saving this as a useful checkpoint.

kensternberg-authentik committed:
Merge branch 'main' into web/bug/search-select-focus-issue

* main: (98 commits)
  outposts: implement general paginator for list API requests (#10619)
  website/scripts/docsmg: switch docsmg from using .env to docsmg.env (#10680)
  web: fix dark theme and theme switch (#10667)
  website/scripts/docsmg: Updated README (#10678)
  core: bump goauthentik.io/api/v3 from 3.2024061.8 to 3.2024061.9 (#10669)
  web: bump @babel/preset-env from 7.24.8 to 7.25.0 in /web in the babel group across 1 directory (#10670)
  web: bump the rollup group across 1 directory with 3 updates (#10672)
  web: bump @swc/core from 1.7.2 to 1.7.3 in /web/sfe (#10673)
  web: bump rollup from 4.19.0 to 4.19.1 in /web/sfe (#10674)
  website/scripts/docsmg: Bug fixes (#10668)
  website/integrations: add engomo (#10538)
  ci: fix expression for outpost GHA caching
  website/docs: fix typos in passwordless setup guide (#10661)
  website/docs: fix missing mermaid theme (#10665)
  website/scripts/docsmg: Add install script and readme draft (#10659)
  web: bump API Client version (#10653)
  website/scripts: Add docsmg migration tool (#10658)
  core: add primitives for source property mappings (#10651)
  root: make pyrad a standard dependency (#10649)
  core: remove deprecated sentry-sdk method usage (#10648)
  ...

kensternberg-authentik committed:
.

kensternberg-authentik committed:
Merge branch 'dev' into web/bug/search-select-focus-issue

* dev:
  website/docs: update nginx docs for embedded outposts (#10422)
  root: dependency maintenance (#10689)
  website/integrations: fix Nextcloud scopes (#10688)
  website/docs: explain guarantees around blueprint ordering (#10595)
  core: bump goauthentik.io/api/v3 from 3.2024061.9 to 3.2024061.10 (#10684)
  web: bump @spotlightjs/spotlight from 2.1.1 to 2.2.0 in /web in the sentry group across 1 directory (#10685)
  web: bump the babel group across 1 directory with 2 updates (#10686)
  web: bump API Client version (#10681)
  sources/scim: add property mappings (#10650)

kensternberg-authentik committed:
Merge branch 'main' into web/bug/search-select-focus-issue

* main:
  root: only load daphne django app in development (#10723)
  web: fix theme not applying to document correctly (#10721)
  stages/authenticator_webauthn: Update FIDO MDS3 & Passkey aaguid blobs (#10712)
  core: bump sentry-sdk from 2.11.0 to 2.12.0 (#10713)
  core: bump uvicorn from 0.30.3 to 0.30.4 (#10714)
  web: bump @sentry/browser from 8.20.0 to 8.21.0 in /web in the sentry group across 1 directory (#10715)
  web: bump @babel/preset-env from 7.25.2 to 7.25.3 in /web in the babel group across 1 directory (#10716)
  core: bump goauthentik.io/api/v3 from 3.2024061.10 to 3.2024062.1 (#10717)
  web: bump @swc/core from 1.7.3 to 1.7.4 in /web/sfe (#10719)
  web: bump API Client version (#10711)
  release: 2024.6.2
  website/docs: add 'Standalone' tap to Paperless-ngx config (#10657)
  website/docs: integrations: add linkwarden integration (#10683)
  providers/saml: remove redundant ACSUrl in POST (#10707)
  providers/radius: fix custom attribute (#10704)
  website/docs: prepare 2024.6.2 (#10697)
  outposts: ensure minimum refresh interval (#10701)
  core: bump google-api-python-client from 2.138.0 to 2.139.0 (#10691)
  web: bump @spotlightjs/spotlight from 2.2.0 to 2.2.2 in /web in the sentry group across 1 directory (#10692)
  web: bump chromedriver from 127.0.0 to 127.0.1 in /tests/wdio (#10693)

BeryJu committed:
root: insert daphne app in correct order

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

kensternberg-authentik committed:
Merge branch 'main' into web/bug/search-select-focus-issue

* main:
  root: insert daphne app in correct order

kensternberg-authentik committed:
web: implement ak-list-select

Creates a new element, ak-list-select, which is a scrollable list that reports when an element is clicked or
selected by the keyboard.

I was hideously over-engineering ak-search-select-menu, and I decided to try something simpler.  This is
that something.  The events we care about are just "change" and "lost focus", and both of those can be
attached by the parent regardless of portaling.

kensternberg-authentik committed:
web: ak-list-select is complete

An extraction of the "menu" and "list" features from SearchSelect
and DualSelect, this is a very simplified version of a visible list
that emulates the Radio/Select behavior (i.e only one from the
collection may be "valued" at the time).  It has no visible indicators
of selection (aside from some highlighting), as it's meant to be
used to present the list rather than be indicative of any state of
the list.

I was seriously over-engineering the menu.  It turns out, it's just
not that difficult after all.  The only things we care about, really,
are "did the user change the selection," "did the user click out
of the list," and "did the user press the escape key."  Those are
pre-existing events (click w/value, blur, and keydown w/keycode,
respectively), so there was no need for me to introduce new custom
events to handler them.

kensternberg-authentik committed:
web: downgrade sonarjs again, because dependabot

Dammit, really need to tell that machine to leave our versions alone.

kensternberg-authentik committed:
web: search select

After a lot of testing and experimenting, it's finally starting to look stable.
What a pain in the neck this has all been.

kensternberg-authentik committed:
web: hold

kensternberg-authentik committed:
Merge remote-tracking branch 'refs/remotes/origin/main'

* refs/remotes/origin/main: (37 commits)
  release: 2024.6.3
  website/docs: prepare 2024.6.3 release notes (#10775)
  website/scripts: updated readme, added docsmg.env file  (#10710)
  web: bump API Client version (#10777)
  root: fix opencontainers ref (#10776)
  core: applications api: properly select provider (#10373)
  root: remove warnings (#10774)
  enterprise/rac: fix error when listing connection tokens as non-superuser (#10771)
  core: bump golang.org/x/oauth2 from 0.21.0 to 0.22.0 (#10754)
  core: bump goauthentik.io/api/v3 from 3.2024062.1 to 3.2024062.2 (#10753)
  core: bump golang.org/x/sync from 0.7.0 to 0.8.0 (#10755)
  web: bump the rollup group across 1 directory with 3 updates (#10756)
  web: bump core-js from 3.37.1 to 3.38.0 in /web (#10757)
  web: bump @swc/core from 1.7.4 to 1.7.6 in /web/sfe (#10758)
  web: bump rollup from 4.19.2 to 4.20.0 in /web/sfe (#10759)
  core: bump black from 24.4.2 to 24.8.0 (#10760)
  core: bump msgraph-sdk from 1.5.3 to 1.5.4 (#10761)
  core: bump coverage from 7.6.0 to 7.6.1 (#10762)
  core: bump ruff from 0.5.5 to 0.5.6 (#10763)
  core: bump django-filter from 24.2 to 24.3 (#10764)
  ...

kensternberg-authentik committed:
Merge branch 'main' into web/bug/search-select-focus-issue

* main: (37 commits)
  release: 2024.6.3
  website/docs: prepare 2024.6.3 release notes (#10775)
  website/scripts: updated readme, added docsmg.env file  (#10710)
  web: bump API Client version (#10777)
  root: fix opencontainers ref (#10776)
  core: applications api: properly select provider (#10373)
  root: remove warnings (#10774)
  enterprise/rac: fix error when listing connection tokens as non-superuser (#10771)
  core: bump golang.org/x/oauth2 from 0.21.0 to 0.22.0 (#10754)
  core: bump goauthentik.io/api/v3 from 3.2024062.1 to 3.2024062.2 (#10753)
  core: bump golang.org/x/sync from 0.7.0 to 0.8.0 (#10755)
  web: bump the rollup group across 1 directory with 3 updates (#10756)
  web: bump core-js from 3.37.1 to 3.38.0 in /web (#10757)
  web: bump @swc/core from 1.7.4 to 1.7.6 in /web/sfe (#10758)
  web: bump rollup from 4.19.2 to 4.20.0 in /web/sfe (#10759)
  core: bump black from 24.4.2 to 24.8.0 (#10760)
  core: bump msgraph-sdk from 1.5.3 to 1.5.4 (#10761)
  core: bump coverage from 7.6.0 to 7.6.1 (#10762)
  core: bump ruff from 0.5.5 to 0.5.6 (#10763)
  core: bump django-filter from 24.2 to 24.3 (#10764)
  ...

kensternberg-authentik committed:
web: search select with focus and progressive search

- New component: ak-list-select, which allows you to select from a list of elements, with keyboard
  control.
- New component: ak-portal, which manages elements by moving "slotted" content into a distant
  component, usually one attached to the body, and positions it relative to an existing element.
- ak-search-select-view has been revamped to handle focus, change, input, and blur using
  the browser native event handlers, rather than inventing my own.
- ak-search-select has been turned into a simple driver that manages the view.
- ak-search-select has a new declarative syntax for the most common use case.

I seriously over-engineered this thing, leaning too heavily on outdated knowledge or assumptions
about how the browser works.  The native event handlers attached at the component's borders works
more than fine, and by attaching the event handlers to the portaled component before sending it
off to the slots, the correct handlers get the message.  This revision leverages the browser
a *lot* more, and gets much more effective interaction with much less code.

`<ak-list-select>` is a new component that replaces the ad-hoc menu object of the old SearchSelect.
It is a standalone component that just shows a list, allows someone to navigate that list with the
keyboard or the mouse. By default, it is limited to half the height of the viewport.

The list does not have an indicator of "selected" at this time.  That's just a side effect of it
being developed as an adjunct to search-select.  Its design does not preclude extension.

It has a *lot* of CSS components that can be customized. The properties and events are documented,
but there is only one event: `change`. Consistent with HTML, the value is not sent with the `change`
event; clients are expected to extract it with `change:event.target.value`.

Like all HTML components, it is completely stringly defined; the value is either a string or
undefined.

`<ak-portal>` is a somewhat specialized "portal" component that places an `ak-list-select` in an
object on top of the existing DOM content. It can generalized to do this with any component, though,
and can be extended. It has no events or CSS, since it's "just" managing the portaling relationship.

`<ak-search-select-view>` is the heart of the system.  It takes a collection options and behaves
like an autocomplete component for them.  The only unique event it sends out is `change`, and like
`ak-list-select`, it expects the client to retrieve the value.

Like all HTML components, it is completely stringly defined; the value is either a string or
undefined.

This is the SearchSelect component we've all known to come and love, but with a better pop-up and
cleaner keyboard interaction.  It emits only one event, `ak-change`, which *does* carry the value
with it.

The Storybooks have been updated to show the current version of Search Select, with a (simulated)
API layer as well as more blunt stringly-typed tests for the View layer.  A handful of tests have
been provided to cover a number of edge cases that I discovered during testing.  These run fine
with the `npx` command, and I would love to see them integrated into CI/CD.

The search select fields `renderElement`, `renderDescription`, and `value` properties of
`ak-search-select` have been modified to take a string.  For example, the search for the
list of user looks like this:

```
<ak-search-select
    .fetchObjects=${async (query?: string): Promise<User[]> => {
        const args: CoreUsersListRequest = { ordering: "username" };
        if (query !== undefined) {
            args.search = query;
        }
        const users = await new CoreApi(DEFAULT_CONFIG).coreUsersList(args);
        return users.results;
    }}
    .renderElement=${(user: User): string => {
        return user.username;
    }}
    .renderDescription=${(user: User): TemplateResult => {
        return html`${user.name}`;
    }}
    .value=${(user: User | undefined): string | undefined => {
        return user?.username;
     }}
></ak-search-select>
```

The most common syntax for the these three fields is "just return the string contents of a field by
name," in the case of the description wrapped in a TemplateResult with no DOM components. By
automating that initialization in the `connectedCallback` of the `ak-search-select` component,
this object would look like:

<ak-search-select
    .fetchObjects=${async (query?: string): Promise<User[]> => {
        const args: CoreUsersListRequest = { ordering: "username" };
        if (query !== undefined) {
            args.search = query;
        }
        const users = await new CoreApi(DEFAULT_CONFIG).coreUsersList(args);
        return users.results;
    }}
    .renderElement=${"username"}
    .renderDescription=${"name"}
    .value=${"username"}
></ak-search-select>
```

Due to a limitation in the way properties (such as functions) are interpreted, the syntax
`renderElement="username"` is invalid; it has to be a property expression. Sorry; best I could do.

The old syntax works just fine.  This is a "detect and extend at runtime" enhancement.

kensternberg-authentik commented on authentik/root/settings.py:
```diff
@@ -516,7 +516,7 @@ def _update_settings(app_path: str):
         "rest_framework.renderers.BrowsableAPIRenderer"
     )
-    SHARED_APPS.append("daphne")
+    SHARED_APPS.insert(SHARED_APPS.index("django.contrib.staticfiles"),"daphne")
```
Not sure why this is showing up.  I thought we'd handled this?  @BeryJu?

kensternberg-authentik commented on web/src/elements/ak-dual-select/types.ts:
```diff
@@ -2,9 +2,21 @@ import { TemplateResult } from "lit";
-// missing, it will use the label, which doesn't always work for TemplateResults).
-export type DualSelectPair<T = never> = [string, string | TemplateResult, string?, T?];
+//
+// - key: string
```
This is pure "cleaning up the documentation."  Typescript's "named tuples" are lovely for self-documenting code.

kensternberg-authentik commented on web/package.json:
```diff
@@ -120,7 +120,7 @@
         "eslint-plugin-custom-elements": "0.0.8",
         "eslint-plugin-lit": "^1.11.0",
-        "eslint-plugin-sonarjs": "^1.0.3",
+        "eslint-plugin-sonarjs": "^0.25.1",
```
How do I tell Dependabot to leave our stuff alone?

BeryJu replied:
I guess https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore?

kensternberg-authentik commented on web/scripts/eslint-nightmare.mjs:
```diff
@@ -27,17 +28,29 @@ const eslintConfig = {
             "quotes": ["error", "double", { avoidEscape: true }],
             "semi": ["error", "always"],
             "@typescript-eslint/ban-ts-comment": "off",
-            "sonarjs/cognitive-complexity": ["error", 9],
```
Just some revisions to nightmare mode.  :-)

kensternberg-authentik commented on web/src/admin/common/ak-flow-search/ak-flow-search.stories.ts:
```diff
@@ -61,7 +61,7 @@ const mockData = {
 
 const metadata: Meta<AkFlowSearch<Flow>> = {
-    title: "Elements / Select Search / Flow",
+    title: "Elements / Search Select / Flow",
```
Embarassing!

kensternberg-authentik commented on web/src/admin/providers/ldap/LDAPProviderForm.ts:
```diff
@@ -86,12 +86,8 @@ export class LDAPProviderFormPage extends WithBrandConfig(BaseProviderForm<LDAPP
-                        return group?.pk;
-                    }}
+                    .renderElement=${"name"}
+                    .value=${"pk"}
```
My demo that the new shorter declaration scheme works.  And it does!

BeryJu replied:
While I do agree this is much shorter and concise, we loose the compile-time type checking of typescript and I think this could make it very easy to introduce bugs

kensternberg-authentik committed:
Added comments to the Component Driver Harness.

kensternberg-authentik commented on web/src/elements/types.ts:
```diff
@@ -9,3 +10,66 @@ export type Constructor<T = object> = new (...args: any[]) => T;
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type AbstractConstructor<T = object> = abstract new (...args: any[]) => T;
+
```
One thing that... irritates... me is that ak-dual-select does *not* adhere to this protocol.  It would be nice if all of our search-and-select tools did.  It's a YAGNI right now, but it's on my mind.

kensternberg-authentik reviewed (commented):
Just leaving roadmarks.

kensternberg-authentik committed:
Added more safety and comments.

kensternberg-authentik committed:
Merge branch 'main' into web/bug/search-select-focus-issue

* main:
  blueprints: handle model referencing non-existent app/model (#10796)
  website/docs: add more content about flows (#10527)
  brands: add OIDC webfinger support (#10400)
  web/admin: fix selectable card colour in dark theme (#10794)
  web: bump API Client version (#10793)
  policies: add GeoIP policy (#10454)
  core: bump debugpy from 1.8.3 to 1.8.5 (#10781)
  web: bump @sentry/browser from 8.22.0 to 8.23.0 in /web in the sentry group across 1 directory (#10782)
  website: bump postcss from 8.4.40 to 8.4.41 in /website (#10783)
  web: bump the wdio group across 2 directories with 4 updates (#10785)
  web: bump @lit/localize from 0.12.1 to 0.12.2 in /web (#10786)
  web: bump @floating-ui/dom from 1.6.8 to 1.6.9 in /web (#10787)
  web: bump @lit/localize-tools from 0.7.2 to 0.8.0 in /web (#10788)
  web: bump lit from 3.1.4 to 3.2.0 in /web (#10789)
  core: bump goauthentik.io/api/v3 from 3.2024062.2 to 3.2024063.1 (#10790)
  web: bump API Client version (#10779)

kensternberg-authentik committed:
Merge branch 'main' into web/bug/search-select-focus-issue

* main: (57 commits)
  web/elements: fix empty enterprise banner (#10882)
  root: fix docker build warnings (#10881)
  website/integrations: compress images (#10880)
  website/integrations: add Frappe (#10797)
  translate: Updates for file web/xliff/en.xlf in ru (#10878)
  core, web: update translations (#10877)
  web: bump API Client version (#10876)
  providers: add provider/ prefix for property mappings API (#10874)
  core, web: update translations (#10831)
  core: bump gunicorn from 22.0.0 to 23.0.0 (#10861)
  web: bump the swc group across 2 directories with 11 updates (#10868)
  web: bump ts-pattern from 5.2.0 to 5.3.1 in /web (#10870)
  web: bump @eslint/js from 9.8.0 to 9.9.0 in /web (#10871)
  web: bump @sentry/browser from 8.24.0 to 8.25.0 in /web in the sentry group across 1 directory (#10853)
  core: bump goauthentik.io/api/v3 from 3.2024063.6 to 3.2024063.8 (#10850)
  core: bump paramiko from 3.4.0 to 3.4.1 (#10862)
  core: bump lxml from 5.2.2 to 5.3.0 (#10863)
  core: bump watchdog from 4.0.1 to 4.0.2 (#10864)
  web: bump API Client version (#10844)
  core: bump aiohttp from 3.9.5 to 3.10.2 (#10843)
  ...

kensternberg-authentik committed:
web: remove string-based access to API; replace with a consolidated "adapter" layer.

Clean out the string-based API layer in SearchSelect.  Break SearchSelect into a
"Base" that does all the work, and then wrap it in two different front-ends:
one that conforms to the old WCAPI, and one with a slightly new WCAPI:

```
<ak-search-select-ez
    .config=${{
        fetchObjects: async (query?: string): Promise<Group[]> => {
            const args: CoreGroupsListRequest = {
                ordering: "name",
                includeUsers: false,
            };
            if (query !== undefined) {
                args.search = query;
            }
            const groups = await new CoreApi(DEFAULT_CONFIG).coreGroupsList(
                args,
            );
            return groups.results;
        },
        renderElement: (group: Group): string => group.name,
        value: (group: Group | undefined): string | undefined => group?.pk,
        selected: (group: Group): boolean => group.pk === this.instance?.group
     }}
    blankable
>
</ak-search-select-ez>
```

kensternberg-authentik committed:
Prettier had opinions. In one case, an important opinion.

kensternberg-authentik committed:
Rename test and fix lint error.

BeryJu committed:
Merge branch 'main' into web/bug/search-select-focus-issue

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

# Conflicts:
#	web/src/admin/providers/ldap/LDAPProviderForm.ts

BeryJu committed:
fix lint

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
Merge branch 'main' into web/bug/search-select-focus-issue
